UnityProject/Assets/Scripts/Medical/CloningConsole.cs

```diff
@@ -24,32 +23,45 @@ public void ToggleLock()
 
 	public void Scan()
 	{
-		if(scanner && scanner.occupant)
+		if (scanner && scanner.occupant)
 		{
 			var mob = scanner.occupant;
+			var mobID = scanner.occupant.mobID;
+			var playerScript = mob.GetComponent<PlayerScript>();
+			var mind = playerScript.mind;
+			var name = playerScript.playerName;
+			var characterSettings = playerScript.CharacterSettings;
+			var oxyDmg = mob.bloodSystem.oxygenDamage;
+			var burnDmg = mob.GetTotalBurnDamage();
+			var toxinDmg = 0;
+			var bruteDmg = mob.GetTotalBruteDamage();
 			var uniqueIdentifier = "35562Eb18150514630991";
 			for (int i = 0; i < CloningRecords.Count; i++)
 			{
-				if(uniqueIdentifier == CloningRecords[i].UniqueIdentifier)
+				var record = CloningRecords[i];
+				if (mobID == record.mobID)
 				{
+					record.UpdateRecord(name, oxyDmg, burnDmg, toxinDmg, bruteDmg, uniqueIdentifier, characterSettings, mobID, mind);
 					return;
 				}
 			}
-			var name = mob.GetComponent<PlayerScript>().playerName;
-			var oxyDmg = mob.bloodSystem.oxygenDamage;
-			var burnDmg = mob.GetTotalBurnDamage();
-			var toxinDmg = 0;
-			var bruteDmg = mob.GetTotalBruteDamage();
-			CreateRecord(name, oxyDmg, burnDmg, toxinDmg, bruteDmg, uniqueIdentifier);
+			CreateRecord(name, oxyDmg, burnDmg, toxinDmg, bruteDmg, uniqueIdentifier, characterSettings, mobID, mind);
```

Comment:
might've introduced a local struct to contain and transfer all that info in
instead of 9000 parameters